### PR TITLE
fix sample

### PR DESCRIPTION
**Pull request by wjhuang2016**

### CODE REVIEW

File: src/blob_gc_job.cc [hunk 1]
```diff
@@ -166,55 +166,32 @@ Status BlobGCJob::SampleCandidateFiles() {
...
 
   uint64_t iterated_size{0};
   uint64_t discardable_size{0};
-  for (iter.Next();
-       iterated_size < sample_size_window && iter.status().ok() && iter.Valid();
-       iter.Next()) {
+  for ( ; iter.Valid(); iter.Next()) {
     BlobIndex blob_index = iter.GetBlobIndex();
     uint64_t total_length = blob_index.blob_handle.size;
     iterated_size += total_length;
```
Comment (lhy1024) [inline src/blob_gc_job.cc]: Iterated_size should be offset of record than size.

**Review by lhy1024 — COMMENTED** [review]
And the PurgeBlobs test is failed in blob_gc_job_test.cc.

File: src/blob_gc_job.cc [hunk 2]
```diff
@@ -232,7 +209,7 @@ Status BlobGCJob::DoSample(const BlobFileMeta* file, bool* selected) {
 
   *selected =
       discardable_size >=
-      std::ceil(sample_size_window *
+      std::ceil(iterated_size *
```
Comment (lhy1024) [inline src/blob_gc_job.cc:220]: iterated_size should be offset than size;

Comment (wjhuang2016) [reply src/blob_gc_job.cc:220]: Then the blob file is not likely to gc if it has many holes.

**Comment by wjhuang2016** [comment]
> And the PurgeBlobs test is failed in blob_gc_job_test.cc.

Fixed.

**Review by lhy1024 — APPROVED** [review]
The new sample method should be added to todo list.

**Pull request merged**